cocos/core/renderer/scene/model.ts

```diff
@@ -297,10 +288,20 @@ export class Model {
     public isDynamicBatching = false;
 
     /**
-     * @en The instance attributes
-     * @zh 实例化属性
+     * @en The instance attribute block, access by subModel
+     * @zh 硬件实例化属性，通过子模型访问
      */
-    public instancedAttributes: IInstancedAttributeBlock = { buffer: null!, views: [], attributes: [] };
+    public getInstancedAttributeBlock (subModel: SubModel): IInstancedAttributeBlock {
+        if (!this._instancedAttributes.has(subModel)) {
+            this._instancedAttributes.set(subModel, { buffer: null!, views: [], attributes: [] });
+        }
+        return this._instancedAttributes.get(subModel) as IInstancedAttributeBlock;
+    }
+
+    get instancedAttributes () {
+        return this._instancedAttributes;
+    }
+    protected _instancedAttributes: Map<SubModel, IInstancedAttributeBlock> = new Map<SubModel, IInstancedAttributeBlock>();
```

Comment:
How about rename to `_instancedAttributeMap `, as `_instancedAttributes ` is always used for vector.

Reply:
I think it is better to record the attribute block and world matrix index in SubModel, then can avoid map operation.